### PR TITLE
ci: onboarding files from repo-template-node

### PR DESCRIPTION
**Pull request by gjanco**
bringing common files in template up to date
template: repo-template-node
branch: main
commit: 4c429904672f352d946e9e76e53781b7676f7510

Ref: REL-91

### CODE REVIEW

File: package.json
Comment (esatterwhite) [inline package.json:1]: The project name should be `@mezmo/cloudwatch`. That puts a `cloudwatch` package under our mezmo org namespaces. 
namespaces are the current best practice

**Review by esatterwhite — COMMENTED** [review]
Going with our open source naming conventions the repo should be called `cloudwatch-node`.

Its already under the `mezmo` org, we don't need to include it in the repo name. and the `-<lang>` helps avoid repo naming woes if we build another one for a different language.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,127 @@
+def CREDS = [
+    aws(credentialsId: 'aws',
+        accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+        secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'),
```
Comment (esatterwhite) [inline Jenkinsfile:4]: We can exclude, we aren't doing anything with aws here.

**Comment by esatterwhite** [comment]
We can can just create a little script to run as a `prerelease` / `prerelease:dry` hook to generate the zip file 

```javascript
'use strict'

const {promises: fs} = require('node:fs')
const ADMZip = require('adm-zip');

async function main() {
  await fs.mkdir('dist', {recursive: true})
  const zip = new ADMZip()
  zip.addLocalFolder('files', 'files') // any directory we need to incldue
  zip.addLocalFolder('./node_modules', 'node_modules') // must include node_modules
  zip.addLocalFile('index.js')
  zip.writeZip('dist/cloudwatch.zip')

}

main()
  .then(console.log)
  .catch((err) => {
    process.nextTick(() => {
      throw err
    })
});
```

our release config will attach anything in `dist` to the github release

**Comment by gjanco** [comment]
Thanks for all the feedback, @esatterwhite.  For the build/release stuff I followed [this](https://mezmo.slite.com/app/docs/L9Ix7KlB_sfrDw).  My goal is to have CI and simply create a ZIP in releases (for installing into lambda).  Should I start over from some other reference, or just make the changes you recommend?

File: package.json
Comment (gjanco) [reply package.json:1]: This is not going to be an NPM package, should I still make these changes?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    string(credentialsId: 'github-api-token',
+           variable: 'GITHUB_TOKEN')
```
[the remark this answers is not in the document]
Comment (gjanco) [reply Jenkinsfile:6]: I believe I worked with them to set up their token with the Mezmo org.

Will not be posting to NPM.

**Comment by gjanco** [comment]
Closing this PR in lieu of one that I was more directly controlling: https://github.com/mezmo/mezmo-cloudwatch/pull/4

**Comment by gjanco** [comment]
FYI, @esatterwhite, I took on most of your feedback / cross referenced it with https://github.com/logdna/logdna-s3

**Pull request closed without merging**